sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HivePlanTest.scala

```diff
@@ -49,4 +49,12 @@ class HivePlanTest extends QueryTest with TestHiveSingleton {
     assert(plan.collect{ case w: logical.Window => w }.size === 1,
       "Should have only 1 Window operator.")
   }
+
+  test("SPARK-14580 Hive IfCoercion should preserve predicate") {
+    val plan = sql("SELECT IF(assert_true(false), 1, 2)").queryExecution.analyzed
```

Comment:
can you create a different test case? in general we are going to remove hive specific things so i don't think this is the right place to test this.